### PR TITLE
Implement RovingTabIndexMixin and 2 other mixins

### LINKED ISSUE #4: Implement roving tabindex mixin
There is a part of duplicating logic in different components:

- [vaadin-accordion](https://github.com/vaadin/vaadin-accordion/blob/ae0988c21ddd3e9e2efa5e05cd28661c7c14810d/src/vaadin-accordion.html#L171-L230) 
- [vadin-list-mixin](https://github.com/vaadin/vaadin-list-mixin/blob/683bf046adc0b31ba5aeb9273037995f73ae0949/vaadin-list-mixin.html#L149-L206)

This logic should be extracted to a separate mixin so we can reuse it.

We might include `_vertical` as a protected property, and decouple it from `orientation`. 
So technically it would allow us to have vertical accordion, alongside tabs / list box.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #4 

This PR contains the 3 mixins prototyped at vaadin/vaadin-accordion#30

1. `SlottedItemsMixin`
2. `KeyboardDirectionMixin`
3. `RovingTabIndexMixin`

They are needed for the following components:

- `vaadin-accordion`: 1 and 2 only
- `vaadin-tabs`: 1, 2 3
- `vaadin-list-box`: 1, 2, 3

### CODE REVIEW

File: packages/keyboard-direction-mixin/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# @vaadin/keyboard-direction-mixin
+
+This package provides `KeyboardDirectionMixin` that does the following:
+
+1. Add `keydown` listener on the element
+
+2. Navigate through `items` via arrow keys, <kbd>Home</knd> and <kbd>End</kbd>
```
Comment (Haprog) [inline packages/keyboard-direction-mixin/README.md]: Typo in `Home</knd>` should be `Home</kbd>`

File: packages/slotted-items-mixin/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# @vaadin/slotted-items-mixin
+
+This package provides `SlottedItemsMixin` that does the following:
+
+1. Set read-only `items` property based on the light DOM children
+
+2. Add `slotchange` listener on the default `slot` to update items
+
+3. Dispatch `items-changed` event when `items` value
```
Comment (Haprog) [inline packages/slotted-items-mixin/README.md]: Something is missing at the end. Maybe something like "Dispatch `items-changed` event when `items` value is changed"?

File: packages/keyboard-direction-mixin/README.md (hunk 1 above)
Comment (web-padawan) [reply packages/keyboard-direction-mixin/README.md]: Done.

File: packages/slotted-items-mixin/README.md (hunk 2 above)
Comment (web-padawan) [reply packages/slotted-items-mixin/README.md]: Done.

**Pull request merged**